cvat/apps/engine/serializers.py

```diff
@@ -707,3 +707,81 @@ def create(self, validated_data):
                 models.Comment.objects.create(**comment)
 
         return db_review
+
+class BaseCloudStorageSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.CloudStorage
+        exclude = ['credentials']
+
+class CloudStorageSerializer(serializers.ModelSerializer):
+    owner = BasicUserSerializer(required=False)
+    session_token = serializers.CharField(max_length=440, allow_blank=True, required=False)
+    key = serializers.CharField(max_length=20, allow_blank=True, required=False)
+    secret_key = serializers.CharField(max_length=40, allow_blank=True, required=False)
+    account_name = serializers.CharField(max_length=24, allow_blank=True, required=False)
+
+    class Meta:
+        model = models.CloudStorage
+        fields = (
+            'provider_type', 'resource', 'display_name', 'owner', 'credentials_type',
+            'created_date', 'updated_date', 'session_token', 'account_name', 'key',
+            'secret_key', 'specific_attributes', 'description'
+        )
+        read_only_fields = ('created_date', 'updated_date', 'owner')
+
+    def validate(self, attrs):
+        if attrs.get('provider_type') == models.CloudProviderChoice.AZURE_CONTAINER:
+            if not attrs.get('account_name', ''):
+                raise exceptions.PermissionDenied('Account name for Azure container was not specified')
```

Comment:
@Marishka17 , is it really PermissionDenied error? From my perspective it is just a bad request. Could you please explain?

Reply:
Not all providers require `account_name`, but for Azure it is necessary, I guess I agree with this, **fixed**